src/main/kotlin/com/vanniktech/maven/publish/MavenPublishBasePlugin.kt

```diff
@@ -0,0 +1,26 @@
+package com.vanniktech.maven.publish
+
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.util.VersionNumber
+import org.gradle.api.publish.maven.plugins.MavenPublishPlugin as GradleMavenPublishPlugin
+
+open class MavenPublishBasePlugin : Plugin<Project> {
+
+  override fun apply(project: Project) {
+    val gradleVersion = VersionNumber.parse(project.gradle.gradleVersion)
+    if (gradleVersion < VersionNumber(MINIMUM_GRADLE_MAJOR, MINIMUM_GRADLE_MINOR, MINIMUM_GRADLE_MICRO, null)) {
+      throw IllegalArgumentException("You need Gradle version 6.6.0 or higher")
```

Comment:
Wouldn't this be more of a state thing anyway?

```suggestion
      error("You need Gradle version 6.6.0 or higher")
```

Reply:
6a38ada